stl/inc/xutility

```diff
@@ -5808,32 +5808,30 @@ _NODISCARD _CONSTEXPR20 _FwdIt lower_bound(_FwdIt _First, _FwdIt _Last, const _T
 // FUNCTION TEMPLATE _Swap_ranges_unchecked
 template <class _FwdIt1, class _FwdIt2>
 _CONSTEXPR20 _FwdIt2 _Swap_ranges_unchecked(_FwdIt1 _First1, const _FwdIt1 _Last1, _FwdIt2 _First2) {
-    // swap [_First1, _Last1) with [_First2, ...), no special optimization
-    for (; _First1 != _Last1; ++_First1, (void) ++_First2) {
-        _STD iter_swap(_First1, _First2);
-    }
-
-    return _First2;
-}
+    // swap [_First1, _Last1) with [_First2, ...)
 
 #if _USE_STD_VECTOR_ALGORITHMS
-template <class _Ty, enable_if_t<_Is_trivially_swappable_v<_Ty>, int> = 0>
-_CONSTEXPR20 _Ty* _Swap_ranges_unchecked(_Ty* _First1, _Ty* const _Last1, _Ty* _First2) {
-    // swap [_First1, _Last1) with [_First2, ...), trivially swappable optimization
+    using _Elem1 = remove_reference_t<_Iter_ref_t<_FwdIt1>>;
+    using _Elem2 = remove_reference_t<_Iter_ref_t<_FwdIt2>>;
+    if constexpr (is_same_v<_Elem1, _Elem2> && _Is_trivially_swappable_v<_Elem1> && //
+                  _Iterators_are_contiguous<_FwdIt1, _FwdIt2>) {

```

Comment:
Ditto "wrap before binary operators": (I suspect simply removing the `//` and letting clang-format do its thing will work here)
```suggestion
    if constexpr (is_same_v<_Elem1, _Elem2> && _Is_trivially_swappable_v<_Elem1> //
                  && _Iterators_are_contiguous<_FwdIt1, _FwdIt2>) {
```

Reply:
Without comments looks like this:
```
if constexpr (is_same_v<_Elem1,
                      _Elem2> && _Is_trivially_swappable_v<_Elem1> && _Iterators_are_contiguous<_FwdIt1, _FwdIt2>)
```

Reply:
Am I hearing complaints about the infinite wisdom of clang-format?